hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,261 @@
+"""A switch driver for OpenVswitch."""
+import re
+import logging
+import schema
+import os
+import ast
+import subprocess
+import shlex
+
+from hil.model import db, Switch, BigIntegerType, SwitchSession
+from hil.errors import SwitchError
+
+logger = logging.getLogger(__name__)
+
+# Class layout
+# 1. Public methods:
+# 2. Private methods
+# 3. Unimplemented superclass methods.
+
+
+class Ovs(Switch, SwitchSession):
+    """ Driver for openvswitch. """
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/ovs'
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(
+            BigIntegerType, db.ForeignKey('switch.id'), primary_key=True
+            )
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        """Checks input to match switch parameters."""
+        schema.Schema({
+            'username': basestring,
+            'hostname': basestring,
+            'password': basestring,
+        }).validate(kwargs)
+
+    @staticmethod
+    def validate_port_name(port):
+        """ This driver accepts any string as port name."""
+
+# 1. Public Methods:
+
+    def get_capabilities(self):
+        """Provides the features of this switch that HIL supports. """
+        return['Virtual_switch_for_development_Purpose_only']
+
+    def ovs_connect(self, *command_strings):
+        """Interacts with the Openvswitch.
+
+        Args:
+            *command_strings (tuple) : tuple of shell commands required
+                        to make changes to openvswitch
+        Raises: SwitchError
+        Returns: If successful returns None else logs error message
+        """
+        try:
+            for command in command_strings:
+                arg_list = shlex.split(command)
+                subprocess.check_call(arg_list)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError
```

Comment:
maybe do a `SwitchError('Verbose Error Msg here')`? I got this suggestion from the reviews of my other PRs.

Reply:
Next commit fixes it.